tests/Unit/PrestaShopBundle/Command/ListCommandsAndQueriesCommandTest.php

```diff
@@ -0,0 +1,131 @@
+        );
+
+        $this->commandTester = new CommandTester($command);
+
+        parent::setUp();
+    }
+
+    /**
+     * @dataProvider optionsProvider
+     */
+    public function testExecute(array $options, string $result): void
+    {
+        $this->resourceNameCollectionMock->method('create')->willReturn(new ResourceNameCollection());
+
+        $this->resourceMetadataCollectionMock->method('create')->willReturn(new ResourceMetadataCollection(''));
+
+        $this->commandTester->execute($options);
+
+        static::assertEquals($result,
+            $this->commandTester->getDisplay()
+        );
+    }
+
+    public function optionsProvider(): array
+    {
+        return [
+            [
+                [],
+                "1.\nClass: PrestaShop\PrestaShop\Core\Domain\AuthorizationServer\Command\AddApplicationCommand\nType: Command\nAPI: \nCreates application with provided data\n\n2.\nClass: PrestaShop\PrestaShop\Core\Domain\AuthorizationServer\Command\EditApplicationCommand\nType: Command\nAPI: \nEdit application with provided data\n\n3.\nClass: PrestaShop\PrestaShop\Core\Domain\AuthorizationServer\Query\GetApplicationForEditing\nType: Query\nAPI: \nGets application for editing in Back Office\n\n4.\nClass: PrestaShop\PrestaShop\Core\Domain\CustomerService\Command\DeleteCustomerThreadCommand\nType: Command\nAPI: \n\n\n",
+            ],
+            [
+                [
+                    '--domain' => ['CustomerService'],
+                ],
+                "4.\nClass: PrestaShop\PrestaShop\Core\Domain\CustomerService\Command\DeleteCustomerThreadCommand\nType: Command\nAPI: \n\n\n",
+            ],
+            [
+                [
+                    '--format' => 'simple',
+                ],
+                "PrestaShop\PrestaShop\Core\Domain\AuthorizationServer\Command\AddApplicationCommand NOT OK\nPrestaShop\PrestaShop\Core\Domain\AuthorizationServer\Command\EditApplicationCommand NOT OK\nPrestaShop\PrestaShop\Core\Domain\AuthorizationServer\Query\GetApplicationForEditing NOT OK\nPrestaShop\PrestaShop\Core\Domain\CustomerService\Command\DeleteCustomerThreadCommand NOT OK\n",
+            ],
+            [
+                [
+                    '--domain' => ['CustomerService'],
+                    '--format' => 'simple',
+                ],
+                "PrestaShop\PrestaShop\Core\Domain\CustomerService\Command\DeleteCustomerThreadCommand NOT OK\n",
+            ],
+        ];
+    }
+
+    /**
+     * @return string[]
+     */
+    private function getListOfCQRSCommands(): array
+    {
+        return [
+            "PrestaShop\PrestaShop\Core\Domain\AuthorizationServer\Command\AddApplicationCommand",
+            "PrestaShop\PrestaShop\Core\Domain\AuthorizationServer\Command\EditApplicationCommand",
+            "PrestaShop\PrestaShop\Core\Domain\AuthorizationServer\Query\GetApplicationForEditing",
+            "PrestaShop\PrestaShop\Core\Domain\CustomerService\Command\DeleteCustomerThreadCommand",
```

Comment:
You have only tested "happy scenarios". Make sure the Command behaves accordingly with invalid inputs such as
- empty string
- 'FooBar'
- badly formed classmane